enhancements/installer/aws-ingress-nlb-security-group.md

```diff
@@ -0,0 +1,498 @@
+- Proper IAM permissions for the CCM's service account will be required to allow it to create, describe, and delete Security Groups in AWS. This needs to be documented as a prerequisite. (TODO review: maybe we don't need this as IAM permissions would be already granted for CLB [review it])
+
+### Risks and Mitigations
+
+> WIP/TBReviewed
+
+- **Increased complexity in CCM**: Adding security group management to CCM increases its complexity. Mitigation: Focus on a minimal and well-tested implementation, drawing inspiration from the existing CLB security group management logic in CCM.   
+- **Potential for inconsistencies with ALBC**: If users later decide to migrate to ALBC, there might be inconsistencies in how security groups are managed. Mitigation: Clearly document the limitations of this approach and the benefits of using ALBC for more advanced scenarios or a broader range of features.
+- **Maintenance burden**: Maintaining this feature in CCM might become challenging if the upstream cloud-provider-aws project evolves significantly in its load balancer management. Mitigation: Upstream the changes to benefit from community maintenance and align with the long-term strategy for load balancer controllers in OpenShift.
+- **IAM Permissions(?)**: Incorrectly configured IAM permissions for the CCM could lead to failures in creating or managing security groups. Mitigation: Provide clear documentation on the necessary IAM permissions and potentially include checks in the CCM to verify these permissions.
+
+
+### Drawbacks
+
+> WIP/TBReviewed
+
+- The short-term solution requires engineering effort to implement and stabilize the changes in CCM and other components.
+- The extent of changes in CCM may require significant engineering effort from Red Hat to maintain this functionality. To mitigate it, this feature proposes minimum changes without disrupting existing flow.
+- This approach duplicates some functionality that already exists in the AWS Load Balancer Controller.
+
+## Alternatives (Not Implemented)
+
+> TODO/TBD
+
+### **Defaulting to AWS Load Balancer Controller (ALBC) for the default router**:
+
+While ALBC provides more comprehensive support for NLB security groups, this option was deemed out of scope for the initial goal of minimal changes and addressing immediate customer needs for security issues within the existing CCM framework.
+
+Migrating to ALBC would involve more significant architectural changes and potentially impact existing deployments. However, this remains a viable long-term strategy.
+
+Here is an overall effort:
+- There might need to work out how we manage ALBC: ALBO will be used, or CCM will manage ALBC? Neither ALBO nor ALBC is in payload at this time. Moving either one into payload requires migrating from CPaaS to Prow, requiring approval to add a new component to the core payload; When ALBO was created, the messaging was that all new components should be addons, no new components in the core payload.
```

Comment:
i would just say "We will need to work out ..."

Reply:
Good suggestions. Fixed, thanks!